src/uct/ib/rc/accel/rc_mlx5_common.c

```diff
@@ -962,6 +962,28 @@ void uct_rc_mlx5_iface_common_dm_cleanup(uct_rc_mlx5_iface_common_t *iface)
 #endif
 }
 
+#if HAVE_DECL_MLX5DV_CREATE_QP
+void uct_rc_mlx5_common_set_qp_dv_attr(uct_rc_mlx5_iface_common_t *iface,
+                                       struct mlx5dv_qp_init_attr *dv_attr,
+                                       unsigned scat2cqe_dir_mask)
+{
+#if HAVE_DECL_MLX5DV_QP_CREATE_ALLOW_SCATTER_TO_CQE
+    uct_ib_dir_t dir;
+
+    dv_attr->comp_mask |= MLX5DV_QP_INIT_ATTR_MASK_QP_CREATE_FLAGS;
+    ucs_for_each_bit(dir, scat2cqe_dir_mask) {
+        if (iface->super.super.config.max_inl_cqe[UCT_IB_DIR_TX] == 0) {
+            dv_attr->create_flags = MLX5DV_QP_CREATE_DISABLE_SCATTER_TO_CQE;
+            return;
+        }
+    }
```

Comment:
I didn't get this loop. is it just evaluation for ```scat2cqe_dir_mask != 0```? variable dir is not used inside of loop, and in the worst case loop will be repeated few times with same action